micromamba/src/update.cpp

```diff
@@ -8,10 +8,121 @@
 
 #include "mamba/api/configuration.hpp"
 #include "mamba/api/update.hpp"
+#include "mamba/api/channel_loader.hpp"
 
+#include "mamba/core/transaction.hpp"
+#include "mamba/core/context.hpp"
+#include "mamba/core/util_os.hpp"
+#include "version.hpp"
+
+extern "C"
+{
+#include "solv/evr.h"
+#include "solv/pool.h"
+#include "solv/conda.h"
+#include "solv/repo.h"
+#include "solv/selection.h"
+#include "solv/solver.h"
+}
 
 using namespace mamba;  // NOLINT(build/namespaces)
 
+int
+update_self(const std::optional<std::string>& version)
+{
+    auto& config = mamba::Configuration::instance();
+    auto& ctx = mamba::Context::instance();
+    config.load();
+
+    mamba::MPool pool;
+    mamba::MultiPackageCache package_caches(ctx.pkgs_dirs);
+
+    auto exp_loaded = load_channels(pool, package_caches, 0);
+    if (!exp_loaded)
+    {
+        throw std::runtime_error(exp_loaded.error().what());
+    }
+
+    pool.create_whatprovides();
+    std::string matchspec;
+    if (!version)
+    {
+        matchspec = fmt::format("micromamba>{}", umamba::version());
+    }
+    else
+    {
+        matchspec = fmt::format("micromamba={}", version.value());
+    }
+
+    auto solvable_ids = pool.select_solvables(pool.matchspec2id(matchspec), true);
+
+    if (solvable_ids.empty())
+    {
+        if (pool.select_solvables(pool.matchspec2id("micromamba")).empty())
+        {
+            throw std::runtime_error(
+                "micromamba not found in the loaded channels. Add 'conda-forge' to your config file.");
+        }
+        else
+        {
+            Console::instance().print(fmt::format(
+                "\nYour micromamba version ({}) is already up to date.", umamba::version()));
+            return 0;
+        }
+    }
+
+    std::optional<PackageInfo> latest_micromamba = pool.id2pkginfo(solvable_ids[0]);
+    LOG_WARNING << latest_micromamba.value().url;
```

Comment:
This line assumes that `latest_micromamba` have a value, or will throw. Is this the behavior we want here? Or should `latest_micromamba` be checked before logging?

Reply:
This was mostly to quickly hack things together and make sure that the functionality itself works (especially on Windows it's tricky because it's not easy to replace binaries that are in use).